wolfSSL/src/chapter02.md

```diff
@@ -1198,6 +2121,110 @@ Can be defined when building for FreeRTOS (<https://www.freertos.org>). If using
 
 Can be defined when building for the FreeRTOS windows simulator (<https://www.freertos.org>).
 
+#### WOLFSSL_CHIBIOS
+
+Can be defined when building for ChibiOS RTOS.
+
+#### WOLFSSL_CMSIS_RTOS
+
+Can be defined when building for Mbed CMIS-RTOS.
+
+#### WOLFSSL_CMSIS_RTOSv2
+
+Can be defined when building for Mbed CMIS-RTOSv2.
+
+#### WOLFSSL_LWIP_NATIVE
+
+Use in platforms for LWIP native.
+
+#### WOLFSSL_DEOS
+
+You can enable the wolfSSL support for Deos RTOS available [here](https://www.ddci.com/products_deos_do_178c_arinc_653/) using this define.
+
+#### WOLFSSL_ESPIDF
+
+Can be defined when building for ESP-IDF.
+
+#### WOLFSSL_LINUXKM
+
+Use if building for Linux Kernel Module.
+
+#### WORD64_AVAILABLE
+
+Platform with 64-bit CPU registers.
+
+#### WOLFSSL_NUCLEUS_1_2
+
+Use if building for Nucleus 1.2.
+
+#### WOLFSSL_PICOTCP
+
+use if if building with PicoTCP.
+
+#### WOLFSSL_RENESAS_RA6M3G
+
+Use if building with RENESAS RA6M3G.
+
+#### WOLFSSL_RENESAS_RA6M4
+
+use if building with RENESAS RA6M4.
+
+#### WOLFSSL_RIOT_OS
+
+Use if building with RIOT-OS.
+
+#### WOLFSSL_uITRON4
+
+Use if building for uITRON4.
+
+#### WOLFSSL_uTKERNEL2
+
+Use if building with uT-Kernel.
+
+#### WOLFSSL_VXWORKS
+
+Use if building with VxWorks.
+
+#### DEVKITPRO
+
+Use when building for devkitPro.
+
+#### WOLFSSL_VXWORKS_6_x
+
+Used only with implementation for VxWorks 6.x only.
+
+#### WOLFSSL_WICED
+
+Used if building for WICED Studio.
+
+#### FREESCALE_KSDK_FREERTOS
+
+Older name of this is FREESCALE_FREE_RTOS but this is used when building for Freescale KSDK FreeRTOS.
+
+#### FREESCALE_KSDK_MQX
+
+Used when building for Freescale KSDK MQX/RTCS/MFS.
+
+#### FREESCALE_MQX_5_0
+
+Used when building for Freescale Classic MQX version 5.0.
+
+#### WOLFSSL_KEIL_TCP_NET
+
+Configure the TCP stack (`MDK_CONF_NETWORK`). By default uses Keil TCP `WOLFSSL_KEIL_TCP_NET`. Use 0 for none or 2 for user io callbacks.
+
+#### INTEL_GALILEO
+
+Used when configuring ARDUINO and wolfSSL. If building for Intel Galileo platform add: `#define INTEL_GALILEO`
+
+#### HAVE_KEIL_RTX
+
+WolfSSL for MDK-RTX-TCP-FS Configuration.
+
+#### _SOCKLEN_T
```

Comment:
This can be removed. Only used with test.h to make sure `socklen_t` is available.